mifosng-provider/src/integrationTest/java/org/mifosplatform/integrationtests/SystemCodeTest.java

```diff
@@ -266,5 +272,96 @@ public void testCodeValuesNotAssignedToTable() {
+        
+        // create first code value
+        final Integer createCodeValueResponseId1 = (Integer) CodeHelper.createCodeValue(this.requestSpec, this.responseSpec,
+        		createCodeResponseParentId, codeValue1, codeDescription1, codeValue1Position, parentId, CodeHelper.SUBRESPONSE_ID_ATTRIBUTE_NAME);
+
+        // create second code value
+        final Integer createCodeValueResponseId2 = (Integer) CodeHelper.createCodeValue(this.requestSpec, this.responseSpec,
+        		createCodeResponseParentId, codeValue2, codeDescription2, codeValue2Position, parentId, CodeHelper.SUBRESPONSE_ID_ATTRIBUTE_NAME);
+
+        // verify two code values created
+
+        final List<HashMap> codeValuesList1 = (List) CodeHelper.getCodeValuesForCode(this.requestSpec, this.responseSpec,
+        		createCodeResponseParentId, "");
+
+        assertEquals("Number of code values returned matches number created", 2, codeValuesList1.size());
+
+       
+        final String dependentcodeName = Utils.randomNameGenerator("StateList", 5);
+        
+        
+ 
+        final Integer createResponseId = (Integer) CodeHelper.createCode(this.requestSpec, this.responseSpec, dependentcodeName, createCodeResponseParentId,
+                CodeHelper.RESPONSE_ID_ATTRIBUTE_NAME);
+        
+        // verify code created
+        final HashMap newCodeAttributes = (HashMap) CodeHelper.getCodeById(this.requestSpec, this.responseSpec, createResponseId, "");
+
+        Assert.assertNotNull(newCodeAttributes);
+        
+        final String codeValue3 = Utils.randomNameGenerator("Kerala", 5);
+        final String codeValue4 = Utils.randomNameGenerator("LV", 5);
+
+        final int codeValue3Position = 1;
+        final int codeValue4Position = 1;
+
+        final String codeDescription3 = "Description1";
+        final String codeDescription4 = "Description2";
+        
+        // create first code value
+        final Integer createCodeValueResponseId3 = (Integer) CodeHelper.createCodeValue(this.requestSpec, this.responseSpec,
+        		createResponseId, codeValue3, codeDescription3, codeValue3Position,  createCodeValueResponseId1, CodeHelper.SUBRESPONSE_ID_ATTRIBUTE_NAME);
+
+        // create second code value
+        final Integer createCodeValueResponseId4 = (Integer) CodeHelper.createCodeValue(this.requestSpec, this.responseSpec,
+        		createResponseId, codeValue4, codeDescription4, codeValue4Position, createCodeValueResponseId2, CodeHelper.SUBRESPONSE_ID_ATTRIBUTE_NAME);
+        
+        // verify two code values created
+
+        final List<HashMap> codeValuesList2 = (List) CodeHelper.getCodeValuesForCode(this.requestSpec, this.responseSpec,
+        		createResponseId, "");
+
+        assertEquals("Number of code values returned matches number created", 2, codeValuesList2.size());
+        
+        final HashMap codeValueChanges = (HashMap) CodeHelper.updateCodeValue(this.requestSpec, this.responseSpec, createResponseId,
```

Comment:
Please use better naming conventions for the variables used that would represent the actual content it would carry